File: GarageGo-BE/Models/YeucauDichVu.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GarageGo_BE.Models
{
    [Table("YEUCAU_DICHVU")]
    public class YeucauDichVu
    {
        [Key]
        public int MaYeuCau { get; set; }

        [Required]
        [StringLength(100)]
        public string TenKhachHang { get; set; } = string.Empty;

        [Required]
        [StringLength(20)]
        public string SoDienThoai { get; set; } = string.Empty;

        [Required]
        [StringLength(500)]
        public string DiaChi { get; set; } = string.Empty;

        public int MaDV { get; set; }

        [StringLength(1000)]
        public string? GhiChu { get; set; }

        public DateTime NgayYeuCau { get; set; } = DateTime.Now;

        [StringLength(20)]
        public string TrangThai { get; set; } = "Mới"; // Mới, Đang xử lý, Hoàn thành, Hủy

        public DateTime? NgayHen { get; set; } // Ngày hẹn khách chọn
        [StringLength(10)]
        public string? GioHen { get; set; } // Giờ hẹn (08:30, 15:00 ...)

        [StringLength(50)]
        public string? Username { get; set; } // Username của user đã đặt

        [ForeignKey("MaDV")]
        public DichVu? DichVu { get; set; }
    }
}

File: GarageGo-BEJava/src/main/java/com/garagego/model/YeucauDichVu.java
package com.garagego.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "YEUCAU_DICHVU")
public class YeucauDichVu {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MaYeuCau")
    private Integer maYeuCau;

    @Column(name = "TenKhachHang", nullable = false, length = 100)
    private String tenKhachHang;

    @Column(name = "SoDienThoai", nullable = false, length = 20)
    private String soDienThoai;

    @Column(name = "DiaChi", nullable = false, length = 500)
    private String diaChi;

    @Column(name = "MaDV")
    private Integer maDV;

    @Column(name = "GhiChu", length = 1000)
    private String ghiChu;

    @Column(name = "NgayYeuCau")
    private LocalDateTime ngayYeuCau = LocalDateTime.now();

    @Column(name = "TrangThai", length = 20)
    private String trangThai = "Mới";

    @Column(name = "NgayHen")
    private LocalDate ngayHen;

    @Column(name = "GioHen", length = 10)
    private String gioHen;

    @Column(name = "Username", length = 50)
    private String username;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "MaDV", insertable = false, updatable = false)
    private DichVu dichVu;
}
